uniflow-core/src/main/kotlin/io/uniflow/core/flow/ActionReducer.kt

```diff
@@ -36,20 +38,23 @@ class ActionReducer(
         actor.send(action)
     }
 
+    @Suppress("UNCHECKED_CAST")
     private suspend fun reduceAction(action: ActionFlow) {
         UniFlowLogger.debug("$tag - execute: $action")
         val currentState: UIState = uiDataStore.currentState
         try {
-            val onSuccess = action.onSuccess
-            flow<UIDataUpdate> {
-                action.flow = this
-                onSuccess(action, currentState)
-            }
-            .onCompletion {
-                UniFlowLogger.debug("$tag - completed: $action")
-            }
-            .collect { dataUpdate ->
-                uiDataStore.pushNewData(dataUpdate)
+            if (!action.klass.isInstance(currentState)) {
+                enqueueAction(ActionFlow(currentState::class as KClass<UIState>, onSuccess = { sendEvent { UIEvent.BadOrWrongState(currentState) } }, onError = action.onError))
```

Comment:
Not sure if this is the correct way of sending a `UIEvent.BadOrWrongState`.

Reply:
yeah, perhaps we can register the `state` class when we make an action. In case of simple action, no need of particular class. In `actionOn` case, we need it